src/eventstore/EventStoreSubscriber.cs

```diff
@@ -459,30 +313,43 @@ private void ProcessEvent(ResolvedEvent resolvedEvent)
                     return;
                 }
 
-                _checkpoint.Write(resolvedEvent.OriginalEventNumber);
+                _checkpoint.Write(resolvedEvent.OriginalEventNumber.ToInt64());
```

Comment:
Having discussed with @rcknight in teams, we should update the subscribe to throw an exception if checkpointing is enabled, and the event number is > long.Max.

https://teams.microsoft.com/l/message/19:2c1ca90124fc455ea75b289c3ad3c3ec@thread.skype/1669113457851?tenantId=b0c1be47-abd8-448c-a6e0-b1ef65a16eae&groupId=ee74f8db-6a44-4230-9890-0e2277229c1f&parentMessageId=1669113457851&teamName=Corsham%20Science%20-%20Software%20Development%20Team&channelName=General&createdTime=1669113457851&allowXTenantAccess=false

We should update docs around the checkpointing constructors to note this restriction.

Reply:
Added exception